slack-api-client/src/test/java/test_with_remote_apis/methods/BlockKit_Test.java

```diff
@@ -365,4 +370,46 @@ public void richTextMessage() throws Exception {
         );
         assertThat(response.getError(), is(nullValue()));
     }
+
+    @Test
+    public void imageBlockWithSlackFile() throws Exception {
+        loadRandomChannel();
+        MethodsClient client = slack.methods(botToken);
+
+        File file = new File("src/test/resources/slack-logo.gif");
+        FilesUploadV2Response uploadResult = client.filesUploadV2(r -> r
+                .file(file)
+                .title("test")
+                // Share the file in a public channel first
+                .channel(randomChannelId)
+        );
+        assertThat(uploadResult.getError(), is(nullValue()));
+
+        String fileId = uploadResult.getFile().getId();
+        String fileUrl = uploadResult.getFile().getPermalink();
+        com.slack.api.model.File.Shares shares = uploadResult.getFile().getShares();
+        while (shares == null || shares.getPublicChannels() == null || shares.getPublicChannels().isEmpty()) {
+            Thread.sleep(300L);
+            shares = client.filesInfo(r -> r.file(fileId)).getFile().getShares();
+        }
+        // Workaround to delete the previous message to share the file
+        String ts = shares.getPublicChannels().get(randomChannelId).get(0).getTs();
+        ChatDeleteResponse deletion = client.chatDelete(r -> r.channel(randomChannelId).ts(ts));
+        assertThat(deletion.getError(), is(nullValue()));
+        Thread.sleep(500L);
+
+        ChatPostMessageResponse response = client.chatPostMessage(r -> r
```

Comment:
I suppose there is no way to validate this message correctly displays the image?

Reply:
The only approach would be to check if there are the blocks in the message property in the response, but it sounds repetitive to me. Validating whether there is no error code such as invalid_blocks in the response (like the current test code does) should be good enough this time.